lightningd/notification.h

```diff
@@ -37,4 +38,24 @@ void notify_forward_event(struct lightningd *ld,
 			  enum onion_type failcode,
 			  struct timeabs *resolved_time);
 
+struct notification {
+	const char *topic;
+	void (*serialize_payload)(void *src, struct json_stream *dest);
+};
+
+AUTODATA_TYPE(notifications, struct notification);
+
+void notification_call(struct lightningd *ld, const char* topic,
+		       void *payload);
```

Comment:
We can likely also generate a type-safe wrapper for each notification that wraps the association `(topic, payload)` and internally calls `notification_call`.

I'd suggest making this `notification_##topic##_call` or `notify_##topic` and then we can just pass in the payload, instead of having to match topic and payload type.

Reply:
Like the `hook`. Ok, I'll do it :-)